Studies/source/GenStudy.cxx

```diff
@@ -0,0 +1,754 @@
+            }
+
+            if(gen_jets.size() ==1)
+                anaData.Higgs_Pt("merged").Fill(HH_Gen_Event.h_bb->momentum.Pt());
+            else
+                anaData.Higgs_Pt("normal").Fill(HH_Gen_Event.h_bb->momentum.Pt());
+
+            if(gen_jets.size() > 2) {
+                anaData.jets_energy("relative").Fill(HH_Gen_Event.h_bb_others_vis.E()/HH_Gen_Event.h_bb_vis_all.E());
+                anaData.jets_momentum("relative").Fill(HH_Gen_Event.h_bb_others_vis.Pt()/HH_Gen_Event.h_bb_vis_all.Pt());
+            }
+
+            if(gen_jets.size() > 2) {
+                anaData.jets_energy("baryons_and_mesons_relative").Fill(bm_others.E()/bm_all.E());
+                anaData.jets_momentum("baryons_and_mesons_relative").Fill(bm_others.Pt()/bm_all.Pt());
+                anaData.diff_rel_energy().Fill(HH_Gen_Event.h_bb_others_vis.E() - bm_others.E());
+            }
+
+            anaData.n("baryons_mesons").Fill(baryons_plus_mesons.size());
+            anaData.bm_vs_jets().Fill(gen_jets.size(), baryons_plus_mesons.size());
+
+            anaData.n("jets_matches").Fill(b_jet_matches_total.size());
+            anaData.n("bm_matches").Fill(bm_matches_total.size());
+
+            anaData.inv_mass_jets().Fill(HH_Gen_Event.h_bb_vis.M());
+            anaData.inv_mass_jets("bm").Fill((baryons_plus_mesons.at(0)->momentum + baryons_plus_mesons.at(1)->momentum).M());
+            anaData.inv_mass_jets("all").Fill(HH_Gen_Event.h_bb_vis_all.M());
+
+            anaData.pt_tau("1").Fill(HH_Gen_Event.vis_tau[0].Pt());
+            anaData.eta_tau("1").Fill(HH_Gen_Event.vis_tau[0].Eta());
+            anaData.pt_tau("2").Fill(HH_Gen_Event.vis_tau[1].Pt());
+            anaData.eta_tau("2").Fill(HH_Gen_Event.vis_tau[1].Eta());
+
+
+            if(tau_matches_total.size() == 1) {
+                for(size_t tau_index = 0; tau_index < 2; ++tau_index) {
+                    if(tau_matches[tau_index].empty()){
+                        anaData.pt_charged_vis_tau("missing").Fill(HH_Gen_Event.vis_charged_tau[tau_index].Pt());
+                    }
+                }
+            }
+
+            //==========================================================================
+            //===========================OUTPUT FOR NEW NTUPLE==============================
+            //==========================================================================
+            if(b_jet_matches_total.size() == 2 && tau_matches_total.size() ==2) {
+                if(tau_matches.at(0).size() == 1 && tau_matches.at(1).size() == 1 && b_matches.at(0).size() == 1 &&
+                        b_matches.at(1).size() == 1){
+                    anaData.n("2_matches_for_all").Fill(1);
+
+                    (*new_tuple)() = event;
+
+                    for (size_t reco_tau_index = 0; reco_tau_index < event.lep_p4.size(); reco_tau_index++) {
+                        size_t matched_gen_tau = 10;
```

Comment:
I set the default value as size_t 10 because the maps and counters of the matching are set<size_t> and map<size_t, set<size_t>>, same for the b part. If that's changed there're infinite warnings that would be difficult to fix, so is it wrong if is defined as I did?